modules/flowable-idm-engine/src/main/java/org/flowable/idm/engine/impl/IdmIdentityServiceImpl.java

```diff
@@ -212,4 +221,46 @@ public PrivilegeQuery createPrivilegeQuery() {
         return commandExecutor.execute(new GetUsersWithPrivilegeCmd(name));
     }
 
+    @Override
+    public PasswordEncoder getPasswordEncoder() {
+        return passwordEncoder;
+    }
+
+    @Override
+    public void setPasswordEncoder(PasswordEncoder passwordEncoder) {
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    public PasswordSalt getPasswordSalt() {
+        return passwordSalt;
+    }
+
+    @Override
+    public void setPasswordSalt(PasswordSalt passwordSalt) {
+        this.passwordSalt = passwordSalt;
+    }
+
+    private String getSalt() {
+        Object source = passwordSalt.getSource();
+        if (source instanceof String) return (String) source;
+        try {
+            Class<?> aClass = Class.forName("org.springframework.security.authentication.dao.SystemWideSaltSource");
```

Comment:
I see your point. I think we should go for 1 provider, probably commons-codec in the flowable-idm-engine package. Then we can add the Spring security option in the flowable-idm-spring module, so there's an alternative that can be configured in a Spring environment. The other ones, Jasypt and JBcypt could be used in the src/test code to show that alternative implementations are also possible, but they should not be part of the src/main code. Then you don't need reflection.